app/javascript/elm/src/TimeRange.elm

```diff
@@ -0,0 +1,56 @@
+module TimeRange exposing (TimeRange(..), defaultTimeRange, update, viewTimeFilter)
```

Comment:
I have to expose `TimeRange(..)` to be able to create expected values in the `TimeRangeTests`. Or is there a better way to do this?

Reply:
I'd suggest keeping in this module only the data structure `TimeRange`. I think the `view` we could inline in the filters modules (we are going to rm the duplication hopefully).

Also, the `update` is a bit strange cause it's not a normal Elm "update" (ie `update : Msg -> Model -> (Model, Cmd Msg)`). I think the decoder and something like `decodeWithDefault : TimeRange -> Encode.Value -> TimeRange` is enough here.

Reply:
I'd suggest always starting with the concrete code (inlined in the place using it) and then abstracting to another module if and only if there's a reason ;)